File: src/cli/commands/delete.ts
/**
 * `agt delete` — Delete a todo (hard delete).
 */

import type { Command } from "commander";
import { findProject, readConfig } from "../../lib/project.js";
import { loadDoc, saveDoc } from "../../lib/storage.js";
import { deleteTodo } from "../../lib/operations.js";
import { parseTodoRef, findTodoByNumber } from "../../lib/queries.js";
import { error, success } from "../output.js";

export function registerDelete(program: Command): void {
  program
    .command("delete")
    .aliases(["rm"])
    .description("Delete a todo (permanent)")
    .argument("<ref>", "Todo reference (e.g. ABC-1 or 1)")
    .option("--json", "Output as JSON")
    .action(async (ref: string, opts: { json?: boolean }) => {
      const paths = findProject();
      if (!paths) error("Not in an agt project. Run 'agt init' first.");

      const config = await readConfig(paths.configPath);
      let doc = await loadDoc(paths.dataPath);
      if (!doc) error("Project data not found.");

      const num = parseTodoRef(ref, config.prefix);
      if (num === null) error(`Invalid todo reference: "${ref}".`);

      const existing = findTodoByNumber(doc, num);
      if (!existing) error(`Todo ${config.prefix}-${num} not found.`);

      doc = deleteTodo(doc, num);
      await saveDoc(paths.dataPath, doc);

      const todoRef = `${config.prefix}-${num}`;
      if (opts.json) {
        console.log(JSON.stringify({ ref: todoRef, deleted: true }));
      } else {
        success(`Deleted ${todoRef}`);
      }
    });
}

File: agt-rs/crates/agt-cli/src/commands/delete.rs
use anyhow::Result;

use agt_lib::operations;
use agt_lib::queries;

use super::{load_project, parse_ref, save_project};

pub fn run(reference: String) -> Result<()> {
    let (paths, mut doc) = load_project()?;
    let (_, prefix, _, _) = queries::read_project_meta(&doc);
    let num = parse_ref(&reference, &prefix)?;

    operations::delete_todo(&mut doc, num, None)?;
    save_project(&paths, &mut doc)?;

    println!("Deleted {}-{}", prefix, num);
    Ok(())
}
